library/kotlin/src/io/envoyproxy/envoymobile/UpstreamHttpProtocol.kt

```diff
@@ -0,0 +1,9 @@
+package io.envoyproxy.envoymobile
+
+/**
+ * Available upstream http protocols.
+ */
+enum class UpstreamHttpProtocol(internal val stringValue: String) {
+  HTTP1("http1"),
+  HTTP2("http2"),
+}
```

Comment:
Can drop this enum if we just use a single header to force h2 (which is sort of non-standard anyways and only works with known destinations).